Detectors/TOF/base/include/TOFBase/Digit.h

```diff
@@ -88,11 +95,40 @@ class Digit
   Int_t mElectronIndex;    //!/< index in electronic format
   Double_t mCalibratedTime;      //!< time of the digits after calibration (not persistent; it will be filled during clusterization)
   Bool_t mIsProblematic = false; //!< flag to tell whether the channel of the digit was problemati; not persistent; default = ok
+  Int_t mTriggerOrbit = 0;       //!< orbit id of trigger event
+  Int_t mTriggerBunch = 0;       //!< bunch id of trigger event
 
   ClassDefNV(Digit, 1);
 };
 
 std::ostream& operator<<(std::ostream& stream, const Digit& dig);
+
+struct ReadoutWindowData {
+  // 1st entry and number of entries in the full vector of digits
+  // for given trigger (or BC or RO frame)
+  int firstEntry;
```

Comment:
When I tried I got some strange behaviour. Since I was not sure about the correct usage of RangeReference  I switched to int.

What is the convenience to use RangeReference?

Reply:
It has exactly the same functionality (+print method) and performance as you implemented, it is just better to use the same code for similar purposes. What was the strange behaviour you have seen?

Reply:
The size (analogous of nEntries) was returned wrong. I will try again (it may be I did something wrong)

Reply:
Hi Ruben,
I understood what is my problem.
I was using a RangeRefComp<5>

It seems that it uses only 5 bits to define the number of entries.

I would switch to 32 bits to be sure that framework survives in case of a 100% occupancy event (160000 entries).

Reply:
30 bits used since 32 is not allowed (too big value, message in compilation!)

Reply:
@noferini  ``RangeRefComp`` is a special class (only 4B-long) which is useful only if the number of entries is limited and can be packed in a few bytes (the same for the 1st entry). For the general case, use ``RangeReference<int,int>``

Reply:
ok. Thanks